bin/config.js

```diff
@@ -54,46 +28,43 @@ const prepareIcons = (input) => {
   return files
 }
 
-const prepareOutput = (output) => {
-  const result = {
-    ...defaultConfig.output,
-    icons: { ...defaultConfig.output.icons },
-  }
-
-  if (typeof output == 'string') result.path = output
-  else {
-    if (output.path) result.path = output.path
-    if (output.publicPath) result.publicPath = output.publicPath
-
-    if (typeof output.icons == 'boolean')
-      result.icons = { ...defaultConfig.output.icons }
-    else if (typeof output.icons == 'string') {
-      result.icons = {
-        files: /.svg$/.test(output.icons) ? null : output.icons,
-        sprite: appendExtension(output.icons, '.svg'),
-      }
-    } else result.icons = { ...output.icons }
-
-    if (result.icons.files)
-      result.icons.files = path.resolve(result.path, result.icons.files)
-    if (result.icons.sprite)
-      result.icons.sprite = path.resolve(result.path, result.icons.sprite)
-  }
-
-  return result
+const mergeAndPrepareConfigs = (...configs) => {
+  // Create absolute path for icons so they can be merged
+  configs.forEach(({ config }) => {
+    if ((config || {}).icons) config.icons = prepareIcons(config.icons)
+  })
+
+  const mergedConfig = merge(...configs)
+  const {
+    config: { output },
+  } = mergedConfig
+  if (output.icons.files)
+    output.icons.files = path.resolve(output.path, output.icons.files)
+
+  if (output.icons.sprite)
+    output.icons.sprite = path.resolve(output.path, output.icons.sprite)
+
+  return mergedConfig
 }
 
-const prepare = (config) => {
-  const icons = prepareIcons(config.icons)
-  const output = prepareOutput(config.output)
-  const svgo = { ...defaultConfig.svgo, ...config.svgo }
-  return { icons, output, svgo }
+const read = async (filepath) => {
+  const explorer = cosmiconfig('design')
+  // If filepath is not supplied we start search for config
+  // from parent directory,
```

Comment:
You say `parent` here but you actually go 2 levels above resolving `mainProjectDirectory`